nixos/doc/rl-2411: improve wording of macOS version warning (#339587)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2411.section.md

## Highlights {#sec-release-24.11-highlights}

- **This will be the last release of Nixpkgs to support macOS 10.12 (Sierra) to 10.15 (Catalina).**
  Starting with release 25.05, the minimum supported version will be macOS 11 (Big Sur).

>   From that point on, packages may or may not work on older versions of macOS.

Reviewer: ```suggestion
  From that point on, we will provide no guarantees that packages will work on older versions of macOS.
```

Author: Pushed a slightly different tweak.
